tensorflow/core/framework/kernel_def_builder.h

```diff
@@ -39,6 +39,9 @@ class KernelDefBuilder {
   KernelDefBuilder& Device(const char* device_type);
   //  KernelDefBuilder& Device(DeviceType device_type);
 
+  // Required: specify the subdevice type of device this kernel supports
```

Comment:
Please end both sentences with period.  Also, why is this required?

Reply:
the subdevice type will be key of kernel registration and dispatch (to avoid conflict of third party gpu device and first party gpu device), If this is not set, I don't know the default value of the subdevice since it is generated from kernel_def.proto.

Reply:
I believe strings are empty [by default](https://developers.google.com/protocol-buffers/docs/proto3#default).

Reply:
Ok, I will change it , making the empty string as the subdevice type for the first party devices.